### PR TITLE
final pass

### PR DESCRIPTION
**Pull request by mbrandenburger**
Signed-off-by: Marcus Brandenburger <bur@zurich.ibm.com>

### CODE REVIEW

File: text/0000-fabric-private-chaincode-lite.md [hunk 1]
```diff
@@ -658,38 +663,43 @@ The project team currently participates in the new Confidential Computing Consor
...
 
+Next we summarize the key material used by the FPC components in addition to standard Fabric key material.
+
+- `enclave_sk` - Enclave signature key, used for FPC Chaincode endorsements. This key is kept inside the FPC Chaincode Enclave.
+- `enclave_vk` - Enclave signature verification key, used for FPC Chaincode endorsement validation and as enclave identifier. This key is made available through the Enclave Registry among all participants.
+- `chaincode_ek` - Chaincode encryption key, used to encrypt transaction arguments. This key is made available through the Enclave Registry among all participants.
+- `chaincode_dk` - Chaincode decryption key, used to decrypt transaction arguments. This key is kept inside the FPC Chaincode Enclave.
+- `sek` - State encryption (symmetric) key, used to encrypt/decrypt and authenticate the contents of chaincode state stored on the ledger. This key is kept inside the FPC Chaincode Enclave.
+- `return_encryption_key` - Chaincode Response Encryption Key, generated and sent by the client to the FPC Chaincode Enclave for response encryption. A fresh key for every invocation is used.
+
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md:704]: Hmm, these keys are all mentioned in the section above including purpose and where it is created (hence kept). The only difference here is that you add the names. Seems a bit strange to duplicate everything? If you really think we have to name the keys as in the protobufs, why not just add them to the earlier section?

File: text/0000-fabric-private-chaincode-lite.md [hunk 2]
```diff
@@ -247,27 +241,38 @@ This class includes smart contracts which operate on sensitive medical data, or
 The former is illustrated in more details below.
 More detailed information on the constraints and related programming model restrictions plus a corresponding security analysis can be found in a [separate document](https://docs.google.com/document/d/1jbiOY6Eq7OLpM_s3nb-4X4AJXROgfRHOrNLQDLxVnsc/).
 
-The [Rollback Protection Extension](#rollback-protection-extension) to FPC Lite can enable support of an even larger class of applications such as private sealed-bid auctions or secure voting which do have intrinsic requirements of release of sensitive data conditioned on private state, e.g., a sealed bid auction outcome should be private until the auction closure is commited on the ledger.
+<!-- The [Rollback Protection Extension](#rollback-protection-extension) to FPC Lite can enable support of an even larger class of applications such as private sealed-bid auctions or secure voting which do have intrinsic requirements of release of sensitive data conditioned on private state, e.g., a sealed bid auction outcome should be private until the auction closure is committed on the ledger. -->
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md:244]: I would leave this: this helps more clearly defining the class of apps we support ...

File: text/0000-fabric-private-chaincode-lite.md [hunk 3]
```diff
@@ -247,27 +241,38 @@ This class includes smart contracts which operate on sensitive medical data, or
...
+
+Federated Learning, as described in this original article: https://ai.googleblog.com/2017/04/federated-learning-collaborative.html, is a class of decentralized algorithms in which training is performed simultaneously across multiple nodes without the need to share training data.
+It provides some inherent privacy even without Trusted Execution, but with Trusted Execution is far more robust against various attacks in case one of the participants is compromised by an attacker or malicious insider.
+Specifically, FPC’s Trusted Execution Environment can assure that only vetted and unmodified training programs can participate in the algorithm, and that it includes mechanisms to foil possible attacks such as Adversarial Machine Learning.
+It can also keep a hardware-signed record of model and training data provenance that can be used to audit the system in case bad data is introduced.
+Through each training job, a model is produced which does not incorporate any of the training data from which it was created; the model is published to the group without any revelation of the private training data.
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md:257]: Hmm, i don't like the replacement of PATE with the google approach you mention. it has ill (non) defined privacy and just slapping TEEs around doesn't make it well defined. The PATE had a very well-defined security, but required a trusted party which perfectly fits FPC. And the resulting privacy is the best you can get ...

File: text/0000-fabric-private-chaincode-lite.md [hunk 4]
```diff
@@ -224,11 +215,14 @@ Arguably, it might even be better to put it _after_ the architecture as there ar
...
 
 - We do assume that a code running inside a TEE cannot be tampered with or its memory inspected. Similarly, we also require that remote attestation provided by a TEE are authentic and prove that only the code referenced in the attestation could have issued it.
+Therefore, all participants/organizations trust a TEE (in particular, the FPC Chaincode Enclave), which can provide such an attestation, regardless of at which peer/organization the TEE is hosted.
+
+- The TEE (FPC Chaincode Enclave) does not trust the hosting peer. That is, all data received via transaction invocations (e.g., the transaction proposal) or via state access operations (e.g., `get_state`) are considered untrusted.
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md]: This is really also a correlary of above bullet and the second one (similar to the second paragraph of above bullet)?
```suggestion
  However, given above-mentioned trust assumptions on peers, a TEE (FPC Chaincode Enclave) cannot trust the hosting peer. Hence all data received via transaction invocations (e.g., the transaction proposal) or via state access operations (e.g., `get_state`) must be considered untrusted.
```

File: text/0000-fabric-private-chaincode-lite.md (hunk 2 above)
Comment (mbrandenburger) [reply text/0000-fabric-private-chaincode-lite.md:244]: this is part of the last paragraph in the `Use case: Privacy-enhanced Federated Learning on FPC Lite`

File: text/0000-fabric-private-chaincode-lite.md (hunk 4 above)
Comment (mbrandenburger) [reply text/0000-fabric-private-chaincode-lite.md]: updated

**Pull request merged**